controllers/persistent-volume-claim/controller.go

```diff
@@ -67,7 +67,9 @@ func (r *PersistentVolumeClaimReconciler) Reconcile(ctx context.Context, req ctr
 		return ctrl.Result{}, nil
 	}
 
-	if !strings.HasPrefix(*pvc.Spec.StorageClassName, controllers.StorageClassPrefix) {
+	// Skip if StorageClassName does not contain the lvms prefix
+	lvmsPrefix, deviceClass, exists := strings.Cut(*pvc.Spec.StorageClassName, "-")
```

Comment:
If StorageClassName is not set this will panic. Empty StorageClassNames are allowed so this could be problematic without a check.

Reply:
Maybe we should add some smarter way in a separate follow-up issue (e.g. a label or annotation) instead of this prefix thing as an issue to differentiate the correct storageclass in the future. WDYT?

Reply:
I agree. I'm also not a fan of using string operations to figure this out :)

Reply:
There is a parameter in the storageclass specifying the deviceclass: 
`Parameters:            csi.storage.k8s.io/fstype=xfs,topolvm.io/device-class=vg1`.

So, I could get the storageclass object and check the parameter value to obtain the deviceclass. Wdyt? Would it be an overkill?